src/framework/uicomponents/view/dialogview.cpp

```diff
@@ -136,6 +231,18 @@ QRect DialogView::viewGeometry() const
     return QRect(m_globalPos.toPoint(), QSize(contentWidth(), contentHeight()));
 }
 
+QSet<QQuickItem*> DialogView::findButtons() const
+{
+    QSet<QQuickItem*> buttons;
+    QList<FlatButtonModel*> buttonModels = findChildren<FlatButtonModel*>();
```

Comment:
Idea:

Let's create a controlType property for the FlatButton component
`property string controlType: "flat_button"`
and by this property determine that it is a button

I don't like the model for FlatButton because it saves the pointer to the parent and returns it

Reply:
Two other ideas:
- FlatButtonModel doesn't need to _save_ the pointer to the parent, because you can just call `QObject::parent` on it
- Instead of adding a property to FlatButton, you could look at `QMetaObject::className`, i.e. `qObjectThatsPotentiallyAFlatButton->metaObject()->className()`. It will not be equal to FlatButton, but will contain `_FlatButton_`. https://stackoverflow.com/questions/44107916/qobjectfindchildren-for-qml-object-finding